backend/selectgen.ml

```diff
@@ -1070,13 +1071,22 @@ method emit_expr_aux (env:environment) exp ~bound_name :
               self#insert_move_results env loc_res rd stack_ofs;
               set_traps_for_raise env;
               Some (rd, unclosed_regions)
-          | Iextcall ({ ty_args; returns; _} as r) ->
+          | Iextcall ({ func; ty_args; returns; _} as r) ->
               let (loc_arg, stack_ofs) =
                 self#emit_extcall_args env ty_args new_args in
+              let keep_for_checking =
+                not returns &&
+                !current_function_is_check_enabled &&
+                String.equal func "caml_flambda2_invalid"
```

Comment:
Added it to `Cmm` to avoid introducing new dependencies into `selectgen` but happy to move elsewhere if you think there is a better place for it without creating cyclic dependencies.

Reply:
I don't have a strong opinion
about where it should be defined
(middle-end developers may
have a better idea), but it is
important to use the symbolic
constant in `Selectgen` (it looks
like it is currently used only in
`To_cmm_shared`).

Reply:
To_cmm depends on ocamloptcomp, so the former should be able to reference it from e.g. selectgen.mli

Reply:
> but it is
important to use the symbolic
constant in Selectgen (it looks
like it is currently used only in
To_cmm_shared).

oh, it was not intentional. fixed.

Reply:
> To_cmm depends on ocamloptcomp, so the former should be able to reference it from e.g. selectgen.mli 

putting it in Selectgen seems backwards. I was thinking maybe `Cmm_helpers` would be a better place than `Cmm`?